### PR TITLE
chore: Spelling

### PR DESCRIPTION
**Pull request by jsoref**
Misspellings identified by the [check-spelling action](https://github.com/marketplace/actions/check-spelling).

It reported: https://github.com/jsoref/argo-cd/commit/8913353f1ea0372c098bf61dd2776989ac697545#commitcomment-39434442

And it validated that the changes in this PR made it happy: https://github.com/jsoref/argo-cd/commit/5c26c0b53eb0953b94cdecc0b7ffa4177f565b4d

Note: the action is not included with this PR. If you're interested, I'm more than happy to make an additional PR for it.

Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Optional. My organization is added to USERS.md.
* [x] I've signed the CLA and my build is green ([troubleshooting builds](https://argoproj.github.io/argo-cd/developer-guide/ci/)).

### CODE REVIEW

File: cmd/argocd/commands/app.go [hunk 1]
```diff
@@ -569,7 +569,7 @@ func setAppSpecOptions(flags *pflag.FlagSet, spec *argoappv1.ApplicationSpec, ap
 	}
 	if flags.Changed("self-heal") {
 		if spec.SyncPolicy == nil || spec.SyncPolicy.Automated == nil {
-			log.Fatal("Cannot set --self-helf: application not configured with automatic sync")
+			log.Fatal("Cannot set --self-heal: application not configured with automatic sync")
```
Comment (jsoref) [inline cmd/argocd/commands/app.go:572]: This is an output change.

File: examples/dashboard.json [hunk 2]
```diff
@@ -3146,7 +3146,7 @@
           "timeFrom": null,
           "timeRegions": [],
           "timeShift": null,
-          "title": "Git Requets (ls-remote)",
+          "title": "Git Requests (ls-remote)",
```
Comment (jsoref) [inline examples/dashboard.json:3149]: I'm not sure about this

File: ui/src/app/login/components/login.scss [hunk 3]
```diff
@@ -47,7 +47,7 @@ $logoNegativePath: 'assets/images/argo.png';
         padding: 20px 0 80px;
     }
 
-    &__box-outter {
+    &__box-outer {
```
Comment (jsoref) [inline ui/src/app/login/components/login.scss:50]: This is probably a bit scary

**Review by jsoref — COMMENTED** [review]
Corrections were mostly selected by Google Sheets / Google Chrome.

I've done a quick review and think everything is correct, but...

**Comment by jsoref** [comment]
#3635

**Review by jannfis — COMMENTED** [review]
Wow @jsoref, this is awesome! Thank you very much for this work!

I have reviewed all changes, they look good to me. 

The only thought I'm having is, whether this should be such a large change affecting code, documentation and tooling all in one. Although that none of the changes to the code part in this PR should actually affect any logic, I would feel a little better if we could roll back changes to the code if necessary in an unlikely event - without rolling back all the other improvements to the docs.

However, this is not a formal request for change and I'd like to hear other opinions on this - @alexmt @mayzhang2000 @rachelwang20

**Comment by jsoref** [comment]
I could split out just `.md` files, or some other slice, just let me know... Or if you prefer to do that, that's also fine. I'm not attached to any particular method of integrating the changes.

We're starting to use your product, and I'm pretty sure the visual things would drive me nuts :-(.

**Comment by alexmt** [comment]
Awesome change @jsoref! Thanks a lot for it!

I agree with @jannfis. The code changes look very safe but we learned that even very safe changes might cause issues. 

I think .md files in separate PR is good enough.

**Review by alexmt — APPROVED** [review]
LGTM

**Pull request merged**